File: M8/Exemplos/App_eXperience_Java/app/src/main/java/com/elgin/elginexperience/shipay/responses/GetOrderStatusResponse.java
package com.elgin.elginexperience.shipay.responses;

import com.elgin.elginexperience.shipay.models.BuyerInfo;
import com.elgin.elginexperience.shipay.models.OrderItem;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class GetOrderStatusResponse {
    @SerializedName("buyer_info")
    public BuyerInfo buyer_info;

    @SerializedName("created_at")
    public String created_at;

    @SerializedName("external_id")
    public String external_id;

    @SerializedName("items")
    public List<OrderItem> items;

    @SerializedName("order_id")
    public String order_id;

    @SerializedName("paid_amount")
    public float paid_amount;

    @SerializedName("payment_date")
    public String payment_date;

    @SerializedName("pix_psp")
    public String pix_psp;

    @SerializedName("status")
    public String status;

    @SerializedName("total_order")
    public float total_order;

    @SerializedName("updated_at")
    public String updated_at;

    @SerializedName("wallet")
    public String wallet;

    @SerializedName("wallet_payment_id")
    public String wallet_payment_id;
}

File: M8/Exemplos/App_eXperience_Kotlin/app/src/main/java/com/example/e1_elgin_kotlin/shipay/responses/GetOrderStatusResponse.kt
package com.example.e1_elgin_kotlin.shipay.responses

import com.example.e1_elgin_kotlin.shipay.models.BuyerInfo
import com.example.e1_elgin_kotlin.shipay.models.OrderItem
import com.google.gson.annotations.SerializedName

data class GetOrderStatusResponse (
    @SerializedName("buyer_info")
    var buyer_info: BuyerInfo,

    @SerializedName("created_at")
    var created_at: String,

    @SerializedName("external_id")
    var external_id: String,

    @SerializedName("items")
    var items: List<OrderItem>,

    @SerializedName("order_id")
    var order_id: String,

    @SerializedName("paid_amount")
    var paid_amount: Float,

    @SerializedName("payment_date")
    var payment_date: String,

    @SerializedName("pix_psp")
    var pix_psp: String,

    @SerializedName("status")
    var status: String,

    @SerializedName("total_order")
    var total_order: Float,

    @SerializedName("updated_at")
    var updated_at: String,

    @SerializedName("wallet")
    var wallet: String,

    @SerializedName("wallet_payment_id")
    var wallet_payment_id: String
)
